app/models/mixins/retirement_mixin.rb

```diff
@@ -231,15 +231,36 @@ def valid_zone?
     respond_to?(:my_zone) && my_zone.present?
   end
 
+  def system_context_requester
+    if try(:evm_owner_id).present?
+      User.find(evm_owner_id)
+    else
+      $log.info("System context defaulting to admin user because owner of #{name} not set.")
+      User.find_by(:userid => 'admin')
+    end
+  end
+
+  def q_user_info(q_options, requester)
+    if requester.present?
+      if requester.kind_of?(String)
```

Comment:
Does this ever get called with a string?

Reply:
Yup! It can!